### PR TITLE
Rebase upstream 

### PR DESCRIPTION
**Pull request by bnookala**
Rebased upstream/master into my master branch. This may or may not have re-introduced dead code. Please help identify any!

### CODE REVIEW

**Comment by mtarng** [comment]
Wait, is this to bring `CatalystCode:master` to parity with `olgaliak/active-learning-detect:master` ?

File: .gitignore [hunk 1]
```diff
@@ -118,12 +118,3 @@ tag/*.csv
 
 # TF exported graph files
 .pb
-
```
Comment (mtarng) [inline .gitignore]: Think we want to keep these

Comment (andrebriggs) [inline .gitignore]: @bnookala we want this stuff

**Comment by mtarng** [comment]
Based on build results, i think some of the file moving or renaming got messed up: https://dev.azure.com/epicstuff/active%20learning/_build/results?buildId=1571

**Review by andrebriggs — COMMENTED** [review]
@bnookala Most of the changes don't intersect with us. What is the plan after this is merged? When will our changes be merged into Olga's branch?

Also I think you should go though a deployment with your branch for sanity testing.

File: cli/training.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    print(vott_json)
+    return vott_json
+            
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-n', '--num-images', type=int)
+    config = Config.read_config(CONFIG_PATH)
+    args = parser.parse_args()
+    train(config, args.num_images)
```
Comment (samiyaakhtar) [inline cli/training.py]: this class is out of date, we moved it to train_vnext.

File: cli/utils/config.py [hunk 3]
```diff
@@ -98,10 +97,8 @@ def read_config(config_path):
             raise MissingConfigException()
 
         parser = configparser.ConfigParser()
-        parser._interpolation = configparser.ExtendedInterpolation()
         parser.read(config_path)
         return Config.read_config_with_parsed_config(parser)
 
-
```
Comment (samiyaakhtar) [inline cli/utils/config.py]: We also moved this file back to utils/config.py

**Comment by bnookala** [comment]
@mtarng @samiyaakhtar @andrebriggs closed this pr since i made a bunch of mistakes when rebasing - going to try again

**Pull request closed without merging**